irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -35,104 +35,91 @@ namespace iroha {
-                }
-                // if node has voted for the committed block
-                if (hash == current_block_.first) {
-                  // append signatures of other nodes
-                  this->copySignatures(commit_message);
-                  log_->info("consensus: commit top block: height {}, hash {}",
-                             current_block_.second->height(),
-                             current_block_.second->hash().hex());
-                  subscriber.on_next(current_block_.second);
-                  subscriber.on_completed();
-                  return;
-                }
-                // node has voted for another block - load committed block
-                const auto model_hash =
-                    hash_provider_->toModelHash(hash.value());
-                // iterate over peers who voted for the committed block
-                rxcpp::observable<>::iterate(commit_message.votes)
-                    // allow other peers to apply commit
-                    .flat_map([this, model_hash](auto vote) {
-                      // map vote to block if it can be loaded
-                      return rxcpp::observable<>::create<
-                          std::shared_ptr<shared_model::interface::Block>>(
-                          [this, model_hash, vote](auto subscriber) {
-                            auto block = block_loader_->retrieveBlock(
-                                vote.signature->publicKey(),
-                                shared_model::crypto::Hash(model_hash));
-                            // if load is successful
-                            if (block) {
-                              // update the cache with block consensus voted for
-                              consensus_result_cache_->insert(*block);
-                              subscriber.on_next(*block);
-                            } else {
-                              log_->error(
-                                  "Could not get block from block loader");
-                            }
-                            subscriber.on_completed();
-                          });
-                    })
-                    // need only the first
-                    .first()
-                    .retry()
-                    .subscribe(
-                        // if load is successful from at least one node
-                        [subscriber](auto block) {
-                          subscriber.on_next(block);
-                          subscriber.on_completed();
-                        },
-                        // if load has failed, no peers provided the block
-                        [this, subscriber](std::exception_ptr) {
-                          log_->error("Cannot load committed block");
-                          subscriber.on_completed();
-                        });
-              });
+          return visit_in_place(
+              message,
+              [this](const CommitMessage &msg) { return handleCommit(msg); },
```

Comment:
Please create a nested static visitor class instead.

Reply:
What's the profit?

Reply:
No redundant `visit_in_place` with two proxy methods.

Reply:
...will be replaced with redundant visit class with two methods?
I don't notice any profit again, could you describe it?

Reply:
...will be replaced with redundant visit class with two methods?
I don't notice any profit again, could you describe it?